src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -167,29 +171,37 @@ public EquippedItemCell(final FixedRowsListView<Item> aItemView, final Configure
 
         engineRating.setStyle("-fx-pref-width: 4em;");
         engineRating.getSelectionModel().selectedItemProperty().addListener((aObservable, aOld, aNew) -> {
-            if (!engineChangeInProgress && !changeEngine(engineXl, engineRating)) {
+            if (!engineChangeInProgress && !changeEngine(engineType, engineRating)) {
                 engineChangeInProgress = true;
                 engineRating.getSelectionModel().select(aOld);
                 engineChangeInProgress = false;
             }
         });
 
-        engineXl.selectedProperty().addListener((aObservable, aOld, aNew) -> {
-            if (!engineChangeInProgress && !changeEngine(engineXl, engineRating)) {
-                engineChangeInProgress = true;
-                engineXl.setSelected(aOld);
-                engineChangeInProgress = false;
+        engineType.setStyle("-fx-pref-width: 4em;");
+        engineType.getSelectionModel().selectedItemProperty().addListener((aObservable, aOld, aNew) -> {
+            if (!engineChangeInProgress) {
+                if (!changeEngine(engineType, engineRating)) {
+                    engineChangeInProgress = true;
+                    engineType.getSelectionModel().select(aOld);
+                    engineChangeInProgress = false;
+                }
+                else {
+                    regenerateEngineRatingDropDown();
+                }
             }
         });
-        setupEngineRatingDropDown(aPgiMode);
+
+        regenerateEngineRatingDropDown();
```

Comment:
Should be enough if these are in `makeEngineGraphic`.

Reply:
Done.